packages/core/src/payment/strategies/stripe-upe/stripe-upe.ts

```diff
@@ -333,6 +333,13 @@ export interface StripeUPEClient {
         clientSecret: string,
     ): Promise<{ paymentIntent?: PaymentIntent; error?: StripeError }>;
 
+    /**
+     * Retrieve a PaymentIntent using its client secret.
+     */
+    retrievePaymentIntent(
+        clientSecret: string,
+    ): Promise<{ paymentIntent?: PaymentIntent; error?: StripeError }>;
```

Comment:
🍹 It might be worth declaring a new type for `{ paymentIntent?: PaymentIntent; error?: StripeError }` I see it's repeated several times. Something like:

```js
type StripeUpeResult = { paymentIntent?: PaymentIntent; error?: StripeError }
```